src/main/webapp/assets/js/feedScript.js

```diff
@@ -118,8 +121,46 @@ async function onLoad() {
   addPosts(posts);
 
   // Add the embedded map to the page.
+  addMapToPage();
+}
+
+/**
+ * Adds the loading spinner on the map.
+ */
+function addMapSpinner() {
+  const map = document.getElementById('map-info-container');
+  const spinner = document.createElement('div');
+  spinner.setAttribute('id', 'map-spinner');
+  spinner.setAttribute('class', 'spinner');
+  map.appendChild(spinner);
+}
+
+/**
+ * Removes the loading spinner on the map.
+ */
+function removeMapSpinner() {
+  const map = document.getElementById('map-info-container');
+  const spinner = document.getElementById('map-spinner');
+  map.removeChild(spinner);
+}
+
+/**
+ * Tries to add the map to the page. The map URL calls the initMap() function as
+ * its callback, which then positions the map and adds markers. If we are unable
+ * to retrieve the secret for the map, then we display an error to the user.
+ */
+function addMapToPage() {
   getSecretFor('javascript-maps-api').then((key) => {
-    // TODO: If the key returns null, we should show a placeholder div with error text.
+    if (key === null) {
```

Comment:
Just thinking about other possibilities of why the map wouldn't appear but is it also possible for the lat/long coordinates to be invalid?

Reply:
It's unlikely that the lat/long for the center of the map would ever be invalid, as we're using a fixed pool of lat/longs provided by government data. The markers' lat/longs are dynamically provided by the geocoding service, so it's a little more likely that maybe something invalid might come back, but in that case the singular marker just wouldn't appear

Reply:
Gotcha! Makes sense!